### PR TITLE
Add null-safety

### PR DESCRIPTION
**Pull request by harsimranmaan**
See: https://dart.dev/null-safety

### CODE REVIEW

**Comment by harsimranmaan** [comment]
Since null-safety is a backwards incompatible change, this might be a good chance to cleanup the fromString API. Looking forward to a null-safe version soon

**Comment by ziofat** [comment]
Hi, thanks for contributing. It looks good to me. I will do some test.

**Comment by nstrelow** [comment]
Awesome to see this.
Great @harsimranmaan @ziofat

File: example/.dart_tool/package_config.json [hunk 1]
```diff
@@ -2,181 +2,145 @@
   "configVersion": 2,
   "packages": [
     {
-      "name": "archive",
```
Comment (nstrelow) [inline example/.dart_tool/package_config.json:5]: I think this file should be excluded from git.
https://github.com/flutter/flutter/blob/master/.gitignore
https://gist.github.com/Cheesetouched/ace446b9e6bedef76eb4fb7afb1ea164

**Comment by Ludonope** [comment]
Any update on this PR?
This is a great package but right now it's preventing me from migrating to null-safety.

**Pull request merged**